src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const { id, title, completed, userId } = currentTodo ?? {};
```

Comment:
The destructuring assignment `const { id, title, completed, userId } = currentTodo ?? {};` is used here. However, if `currentTodo` is `null` or `undefined`, this will result in `id`, `title`, `completed`, and `userId` being `undefined`. Ensure that this behavior is intended and handled correctly in the component.